torch_geometric/loader/utils.py

```diff
@@ -199,10 +207,43 @@ def filter_custom_store(
 # Input Utilities #############################################################
 
 
+@dataclass
+class SamplingInputNodes:
```

Comment:
A few thoughts:
* `SamplingInputNodes` and `InputNodes` as types have different functionalities, but almost identical names. Thoughts on changing this name (e.g. to something like `SamplingInput`) to increase the edit distance between them?
* Would it be better to have `input_nodes: InputNodes`? I am not sure I understand the need for a separate `Dict[Optional[str], Sequence` here. That would also make the relationship between this class and `InputNodes` more clear.
* Am I correct in my understanding that we will pass `sampling_input_nodes.as_list()` to the DataLoader constructor, and once batches are selected by the torch DataLoader, we will convert them `from_list` back to a dict representation that we then pass to the sampler?